### ISSUE TITLE
sql: user-defined types have the wrong namespace in SHOW CREATE TABLE

### ISSUE THREAD

**Issue by mgartner** [cockroachdb/cockroach#72241]
In the example below, notice how the `m` column has type `public.mytype` when it should be `db1.mytype`.

```
defaultdb> CREATE DATABASE db1;
CREATE DATABASE

defaultdb> CREATE TYPE db1.mytype AS ENUM ('foo');
CREATE TYPE

defaultdb> CREATE TABLE db1.t (m db1.mytype);
CREATE TABLE

defaultdb> SHOW CREATE TABLE db1.t;
   table_name  |                      create_statement
---------------+--------------------------------------------------------------
  db1.public.t | CREATE TABLE public.t (
               |     m public.mytype NULL,
               |     rowid INT8 NOT VISIBLE NOT NULL DEFAULT unique_rowid(),
               |     CONSTRAINT "primary" PRIMARY KEY (rowid ASC),
               |     FAMILY "primary" (m, rowid)
               | )
(1 row)

```

Jira issue: CRDB-11058

**Comment by rafiss** [comment]
This is a rough edge with name resolution and the current state of cross-database-references.

The user-defined type is formatted as:
```
// FQName returns the fully qualified name.
func (u UserDefinedTypeName) FQName() string {
	var sb strings.Builder
	// Note that cross-database type references are disabled, so we only
	// format the qualified name with the schema.
	if u.ExplicitSchema {
		sb.WriteString(u.Schema)
		sb.WriteString(".")
	}
	sb.WriteString(u.Name)
	return sb.String()
}
```

So, to clarify, `public.mytype` is the correct type name if you are currently using `db1` as the current database. `public` is the schema name, not the database name. This is similar to how the table name is shown in the CREATE statement -- `public.t`.

The fully qualified name would be `db1.public.mytype`. I'm not sure if it's safe to update `FQName()` to include the catalog name.

I'm also not sure what we want to do about `tree.TableName` formatting. Confusingly, it is formatted with the catalog name in the `table_name` but not in `create_statement.` Maybe @ajwerner can offer some advice?

**Comment by ajwerner** [comment]
We can change it to be fully qualified I suppose. Maybe we could make the formatting more context dependent based on the current database. Once we follow through on eliminating cross database references, this will be somewhat less confusing.

**Comment by ajwerner** [comment]
The whole `<database>.<relation>` thing is a cockroach invention. In postgres you can't refer to things in different databases and two part names are always `<schema>.<relation>`. We have to keep supporting `<database>.<relation>` as a shorthand for `<database>.public.<relation>` because of backwards compat. It's going to get fun when in 22.2 users can drop the public schema.

**Comment by rafiss** [comment]
This is fixed on master and 22.2.
